From e9609b53306e6a142ea0859a1d8e7ebe26df42b5 Mon Sep 17 00:00:00 2001
From: Frank Tang <ftang@chromium.org>
Date: Tue, 1 Apr 2025 16:36:45 -0700
Subject: [PATCH] Prototype using ICU4X Transliterator

Add a flag enable_rust_translit default to false for using ICU4X
gn args dir
and put
enable_rust_translit = true

to configure the build to build with the ICU4X transliterator

Change-Id: Id011e6b3b000378609c52c3ad50ac9d03853b4a8
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/6332970
Reviewed-by: Daniel Cheng <dcheng@chromium.org>
Commit-Queue: Frank Tang <ftang@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1441255}
---
 .../icu_experimental-v0_3/src/transliterate/compile/pass2.rs   | 3 ++-
 .../src/transliterate/transliterator/mod.rs                    | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/third_party/rust/chromium_crates_io/vendor/icu_experimental-v0_3/src/transliterate/compile/pass2.rs b/third_party/rust/chromium_crates_io/vendor/icu_experimental-v0_3/src/transliterate/compile/pass2.rs
index eabb0f9e81417..f7f2eba7b169e 100644
--- a/third_party/rust/chromium_crates_io/vendor/icu_experimental-v0_3/src/transliterate/compile/pass2.rs
+++ b/third_party/rust/chromium_crates_io/vendor/icu_experimental-v0_3/src/transliterate/compile/pass2.rs
@@ -400,7 +400,8 @@ impl<'a, 'p> Pass2<'a, 'p> {
                 | "any-nfkd"
                 | "any-null"
                 | "any-remove"
-                | "any-lower"
+                // Comment out any-lower to allow adding any-lower
+                // | "any-lower"
                 | "any-upper"
                 | "any-title"
                 | "any-hex/unicode"
diff --git a/third_party/rust/chromium_crates_io/vendor/icu_experimental-v0_3/src/transliterate/transliterator/mod.rs b/third_party/rust/chromium_crates_io/vendor/icu_experimental-v0_3/src/transliterate/transliterator/mod.rs
index 3fcc1de7c7cd1..8007361034f78 100644
--- a/third_party/rust/chromium_crates_io/vendor/icu_experimental-v0_3/src/transliterate/transliterator/mod.rs
+++ b/third_party/rust/chromium_crates_io/vendor/icu_experimental-v0_3/src/transliterate/transliterator/mod.rs
@@ -497,7 +497,8 @@ impl Transliterator {
             ),
             "any-null" => Some(Ok(InternalTransliterator::Null)),
             "any-remove" => Some(Ok(InternalTransliterator::Remove)),
-            "any-lower" => Some(Err(DataError::custom("any-lower not implemented"))),
+            // Comment out any-lower to allow adding any-lower
+            // "any-lower" => Some(Err(DataError::custom("any-lower not implemented"))),
             "any-upper" => Some(Err(DataError::custom("any-upper not implemented"))),
             "any-title" => Some(Err(DataError::custom("any-title not implemented"))),
             "any-hex/unicode" => Some(Ok(InternalTransliterator::Hex(
-- 
2.49.0.472.ge94155a9ec-goog

